### ISSUE TITLE
Room DB instrumentation does not work with custom SQLite open helper

### ISSUE THREAD

**Issue by vinaysshenoy** [getsentry/sentry-android-gradle-plugin#234]
_Android Gradle Plugin:_
**7.0.3**

_Gradle:_
**7.3**

_Optimizer:_
- [ ]  ProGuard
- [X]  R8

_Have you created the sentry.properties file?_
- [X]  Yes -> `app/src/release/resources`
- [ ]  No

_SDKs:_
- [X]  sentry-android -> **5.3.0**
- [ ]  sentry-react-native -> which version?

---
I have the following issue:
Room database instrumentation does not instrument database queries for custom SQLite open helper.

In our project, we package our version of SQLite using https://github.com/requery/sqlite-android and use their open helper factory when setting up Room.

**Steps to reproduce:**
- Add https://github.com/requery/sqlite-android to a project with Sentry enabled as described here: https://github.com/requery/sqlite-android#usage.
- When constructing the Room database, use the `RequerySQLiteOpenHelperFactory` as shown:
```kotlin
return Room.databaseBuilder(appContext, AppDatabase::class.java, "red-db")
        .openHelperFactory(RequerySQLiteOpenHelperFactory())
        .build()
```
- Run the project with Sentry DB instrumentation enabled.

**Actual result:**
- DB instrumentation does not get reported to Sentry.

**Expected result:**
- DB instrumentation should get reported to Sentry. This can be confirmed by commenting out the line that provides a custom factory and building the project and running again.

**Comment by romtsn** [comment]
Thanks a lot for trying the new functionality out, really appreciate that.

Unfortunately, at the moment we only support the standard Room setup (i.e. when it's backed by [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) lib. This is a good enhancement though, we may consider adding a support in the future.

I've updated [docs](https://github.com/getsentry/sentry-docs/pull/4399/files) mentioning your case.

**Comment by vinaysshenoy** [comment]
> Thanks a lot for trying the new functionality out, really appreciate that.
> 
> Unfortunately, at the moment we only support the standard Room setup (i.e. when it's backed by [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) lib. This is a good enhancement though, we may consider adding a support in the future.
> 
> I've updated [docs](https://github.com/getsentry/sentry-docs/pull/4399/files) mentioning your case.

Interesting, I thought the bytecode instrumentation was being done at the open helper level, not at the sqlite level itself. Will keep an eye out if this enhancement ever makes it into the library.

In the meantime, would you be able to suggest workarounds we can do to instrument  our current setup on Sentry?

**Comment by marandaneto** [comment]
@vinaysshenoy you can still https://docs.sentry.io/platforms/android/performance/instrumentation/custom-instrumentation/

**Comment by vinaysshenoy** [comment]
@romtsn Nope, the spans don't show up. The only spans which show are the ones for the Android workmanager database, which are managed on a platform SQLite version.

<img width="1171" alt="Screenshot 2021-11-22 at 11 11 18 AM" src="https://user-images.githubusercontent.com/1978065/142806890-e32b972d-7f92-4fde-9230-01a7ff487fa6.png">

**Comment by romtsn** [comment]
@vinaysshenoy with the next version of the gradle plugin (3.11.0) we'll be automatically instrumenting any class that implements `androidx.sqlite.db.SupportSQLiteOpenHelper.Factory`, including requery.

We'll ship the new version tomorrow, please let us know if the new instrumentation is gonna work for you.

Closed by #502.
